### ISSUE TITLE
It does not allow symfony ^4.0 version packages

### ISSUE THREAD

**Issue by yurtesen** [sonata-project/SonataDatagridBundle#118]
This cause problems for example installing Sonata Admin Bundle, see the [related issue](https://github.com/sonata-project/SonataAdminBundle/issues/4830) for more info.

The current settings in composer.json does not allow ^4.0 symfony packages to be installed and conflict if user already has a ^4.x package installed. The composer.json must be updated. (and no, I won't make a PR for updating 2 lines plus few characters, somebody who has access to the project should update and release it.)

Offending lines:
```
"symfony/dependency-injection": "^2.8 || ^3.2",
"symfony/form": "^2.8 || ^3.2"
```

**Comment by greg0ire** [comment]
Hey! Guess who doesn't use Sonata right now, doesn't need this and isn't going to make a PR either? It's me! By the way, did you know we use PRs for everything? That's because you never now when something will break the test suite, and making a PR is the best way to run the tests on all environments. So in a way, we don't have more access than you do. Please change your attitude and make a PR. https://twitter.com/jkup/status/909887066103676928

**Comment by greg0ire** [comment]
Plus it's unclear what you are talking about, the right constraints seem to already be there: https://github.com/sonata-project/SonataDatagridBundle/blob/4352a2f43e19513890406fa71f60cd15e3b12064/composer.json#L21-L22

**Comment by yurtesen** [comment]
@greg0ire I am already doing work for this software by taking time to report the issue. It is neither my task, nor my responsibility to fix it. Sure, I can provide more information if possible. But that is as far as I can go.

Some random information on a random twitter link on internet does not mean it is correct or best way to behave. You shouldn't trust the information you see on internet because it is easier to implement. Just use your logic to deduce the end result.

* people will stop sending issue reports, problems will not be known, will not be fixed.
* people will stop using the package.
* people will make their own competing packages
* one day a competing package where the developer is interested in fixing problems will be more popular
* offending package will be dead

If you are not using the software, do not respond to issues. It is just not nice if you are trying to make other people stop using the software. 

At least, I got now a sour taste from Sonata bundles and I will look for another project to use or work on own solution. I know it will take longer but I can see that it will save me a lot of time and grief in future if I do not rely on Sonata bundles.

**Back to the problem**, You are looking at the wrong file. When people install the software they face with the `2.3.0` version. There is already a person who submitted bug report to SonataAdmin issue tracker because it is impossible to install! See the released [composer.json](https://github.com/sonata-project/SonataDatagridBundle/blob/2.3.0/composer.json#L21-22). It may be fixed in `master` but the problem is about release versions.

**Comment by greg0ire** [comment]
> @greg0ire I am already doing work for this software by taking time to report the issue. It is neither my task, nor my responsibility to fix it. Sure, I can provide more information if possible. But that is as far as I can go.

It's great that you do that already, but bear in mind that Sonata is a community project. We maintainers are "just" enablers, and those quotes mean that it's already quite a pretty big task. I might end up fixing this myself if when I have the time and it becomes a priority, but that might be too far in the future for your liking.

> But that is as far as I can go.

Why? Aren't you a developer? You said yourself it was " 2 lines plus few characters", so what's so hard?

> people will stop sending issue reports, problems will not be known, will not be fixed.
You're the only one doing it with that attitude, most people are happy to oblige and proceed with the PR.
Other simply tell us they don't have the time, and we then try to fix it for them when we can, or let another member of the community handle it.

> If you are not using the software, do not respond to issues. It is just not nice if you are trying to make other people stop using the software.

I'm using it. Just not "currently", but I have projects I will have to migrate in the future. Not trying to stop anyone, look at my Github activity, I'm actually spending a lot of time on Sonata, so what you're saying is pretty silly.

> At least, I got now a sour taste from Sonata bundles and I will look for another project to use or work on own solution. I know it will take longer but I can see that it will save me a lot of time and grief in future if I do not rely on Sonata bundles.

Well in that case, should I help you with your issue or not?

> You are looking at the wrong file.

I linked to the 2.x version, not the master one. So what you need is a new release, there is no need for any PR. I'm going to ask for a release because that's the part you don't have access to.

**Comment by greg0ire** [comment]
Someone already asked for it Sunday, waiting for @Soullivaneuh now.

**Comment by Soullivaneuh** [comment]
@yurtesen Here is your needed release to breath again: https://github.com/sonata-project/SonataDatagridBundle/releases/tag/2.3.1

And BTW, I'm very sorry to having a life and not getting paid for what I'm doing. Really, you are right. We should work harder for free and allow people like you to bawl out on us.

Regards.

**Comment by yurtesen** [comment]
@Soullivaneuh  , I made an issue report. I did NOT tell you fix it now, hurry or do you work etc. I did NOT complain that you are not fixing it. I think there is a misunderstanding.

You are perfectly right that you have no obligation to do anything and I never blamed anybody for not doing anything. But you have to understand that I have even less reason to do something. I am just a guy who is trying to install the software and it does NOT install. I simply wanted to help the developer by reporting the issue, which now I regret that I even did.

I complain because @greg0ire tell me to make PR, now on 2 out of 2 issues I reported. Not to mention, this specific issue simply needed a release. I am willing to help to debug the issues and send information. But my help will end there for software which I don't even know if I will use or not in the long run.

In either case, I hope now everything is clear. I will simply wont use Sonata if the developer attitude is that the development and bugfix of Sonata should be done through users making PRs. Because such software has no future for sure.

**Comment by core23** [comment]
> I will simply wont use Sonata if the developer attitude is that the development and bugfix of Sonata should be done **through users making PRs.**

Who should fix the software in your opinion? Have you ever heard of Open Source software? 

If you want us to fix *every* bug, please pay us!

**Comment by jordisala1991** [comment]
I also prefer that you don't use the software with that attitude.. less problems for us having to deal with our limited time with you instead of improving Sonata.

**Comment by yurtesen** [comment]
@core23 First of all that is not even the issue, your logic is flawed. I did not tell/force anybody to fix anything. I simply reported an issue with the software and said I won't make a PR for this. `greg0ire` told me that I should fix it and make PR. If you think I should pay somebody to fix this issue, with your own logic shouldn't he pay me to do this then? Look at this from my point of view. I already used my time, for free, to report the issue to help this project. It is not my problem if somebody fixes it or not. You shouldn't try to make me fix it . Because with your own explanation, I have no obligation to fix it!

@jordisala1991 There is nothing wrong with my attitude if you understand the problem. I simply reported an issue/problem. But `greg0ire` badgered me by saying that I should make PR and inferred that I am doing something wrong. It is not my obligation to do anything for this code, not even report an issue/problem. I did try to help and I was made to regret it by `greg0ire`.

**Comment by greg0ire** [comment]
> Look at this from my point of view. I already used my time, for free, to report the issue to help this project. It is not my problem if somebody fixes it or not. You shouldn't try to make me fix it . Because with your own explanation, I have no obligation to fix it!

Ok so let's say this was a misunderstanding, I wouldn't want us to continue the bashing forever. There are some things I would like to clarify:

1. No, you have indeed no obligation to fix it at all.
2. Neither do we.
2. The time you spend reporting the issues you find is appreciated.
3. We hope you appreciate the fact that your reports are getting quick attention.
3. Whenever someones report an issue, I try to get them to fix it, because the person is already reproducing the bug.
1. It's completely ok to refuse to make a PR. If you give reasons when refusing though, think carefully, because:
   1. It might be easier for us, but we might be busier than you are with all the other issues, especially if everyone refuses to make a PR, so in the end, yes it is easier, but we have to do it at 3 a.m.
   2. We don't have more access than you do, we go through PRs too.

In conclusion, if you don't want to make a PR, just say so, plenty of contributors do that / simply say they don't have the time, and it's ok with us every time, but you can't hate me for asking.

If one day you want to get one step further in giving back to open source, which yes, you already do through issues, I am here to help if you want to make pull requests.

**Comment by Soullivaneuh** [comment]
@yurtesen we are sorry if you understood @greg0ire asking to be an obligation to fix it. It is not, @greg0ire gave a good resume on his last comment.

Nobody is forcing to fix anything, we are just **proposing** to give fix to make things quicker. Please also note we got the maintenance or sonata-project later than it creation and we don't necessarily use all the available bundles.

So no need to be rude on the issue, telling things like "and no, I won't make a PR ...". You reported the thing, it's great and thank you for that. We ask you to make a PR and you don't want ? Fine, just say no or neither respond to the issue, it's also fine.

Now everything is being said, I lock the issue. Issues is for report and are referenced, there is not place for such debate here.

Regards.
